File: 392. Is Subsequence/Solution.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isSubsequence = function(s, t) {
    let s_ind = 0;
    let t_ind = 0;
    while (t_ind < t.length && s_ind < s.length){
        if (t[t_ind] == s[s_ind]){
            s_ind++;
        }
        t_ind++;
    }
    return s.length == s_ind
};

File: 392. Is Subsequence/Solution.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        t_ind = s_ind = 0
        while t_ind < len(t) and s_ind < len(s):
            if t[t_ind]==s[s_ind]:
                s_ind+=1
            t_ind+=1
        return s_ind == len(s)
